### ISSUE TITLE
Mrtransform with flipped data

### ISSUE THREAD

**Issue by scott-trinkle** [MRtrix3/mrtrix3#1669]
I have calculated a spatial transform from the b0 of image A to the b0 of image B using ANTs and I want to use mrtransform to apply the spatial transform to the corresponding ODF SH coefficients. 

The data in image B are flipped along the x dimension with respect to image A. This is accurately represented in the sform in the nifti file. 

When the spatial transformation is applied to the b0s, ANTs accounts for the flip in x-dimension, so the warped image A is flipped and saved with a new sform (identical to that of image B). 

But when I apply the same transforms to the SH images using MRtrix, the data are not flipped when they are saved/stored. So the warped image A coefficients are transformed accurately, but they are saved with the same sform as the original image A. 

So when visualizing the ODFs in mrtrix for example, mrview takes care of the sforms and everything is aligned. However, if I try to load the data into python for quantitative comparisons, the data are flipped and the comparisons are off. 

Is there a command line flag I'm missing? Effectively it comes down to the fact that mrtransform seems to not be using the sform of the reference image. 

Here is how I am applying the transforms with mrtrix:
```
moving=image_A_sh.nii.gz
reference=image_B_b0.nii.gz
warp=image_A_b0_to_image_b_b0_warp.nii.gz
aff=image_A_b0_to_image_b_b0_aff.mat
output=image_A_sh_warped.nii.gz
```
Generate identity warp using moving image
```
warpinit $moving identity_warp[].nii
```

Compute MRtrix-compatible warp
```
for i in {0..2};
do
    WarpImageMultiTransform 3 identity_warp${i}.nii mrtrix_warp${i}.nii -R $reference $warp $aff;
done
```

Correct the MRtrix warp
```
warpcorrect mrtrix_warp[].nii mrtrix_warp_corrected.mif
```

Apply!
```
mrtransform -interp linear -warp mrtrix_warp_corrected.mif -modulate $moving $output
```

**Comment by Lestropie** [comment]
Hi Scott,

I *think* what you're looking for is the `-template` option in the final `mrtransform` call. That *should* ensure that the output image has an identical voxel grid to whatever template image you specify.

I the absence of use of that option, pretty sure the output image created by `mrtransform` will be defined on the voxel grid of the non-linear warp image; that in turn will depend on the internal operation of the `WarpImageMultiTransform` command.

Rob

**Comment by maxpietsch** [comment]
We default to the sform if sform and qform match to numerical precision but default to the qform [otherwise](https://github.com/MRtrix3/mrtrix3/issues/1318). The ANTs wiki [states](https://github.com/ANTsX/ANTs/wiki/How-does-ANTs-handle-qform-and-sform-in-NIFTI-1-images%3F) that they always use the qform and ignore or overwrite the sform so there should be no mismatch.

After transformation with mrtransform, sform and qform should be very similar. Can you conform that? If not, what is the output of `fslhd $output | grep _xyz`?

I suspect the problem here is your interpretation of sform / qform and the data loading might not take [strides](https://www.biorxiv.org/content/10.1101/551739v1) properly into account. (@dchristiaens had issues with numpy 1.13 handling negative strides wrong which were resolved in numpy 1.15). To test that, you can change the stride order of all images to match sign and order with `mrconvert -strides`. To inspect the changed strides and corresponding transformations, you can use `mrinfo image -norealign`. Note that `mrtransform <input> -template <template>` matches the image grid of <input> to that of <template> but does not modify the image strides.

**Comment by scott-trinkle** [comment]
Thank you both! I added: "-template $reference -strides $reference" to the call to mrtransform and that did the trick. 

Really appreciate the help!
